client/extensions/woocommerce/state/settings/shipping/zones/reducer.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * External dependencies
+ */
+import { isEqual, unionWith, differenceWith, isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	WOOCOMMERCE_SHIPPING_ZONE_ADD,
+	WOOCOMMERCE_SHIPPING_ZONE_CANCEL,
+	WOOCOMMERCE_SHIPPING_ZONE_CLOSE,
+	WOOCOMMERCE_SHIPPING_ZONE_EDIT,
+	WOOCOMMERCE_SHIPPING_ZONE_LOCATION_ADD,
+	WOOCOMMERCE_SHIPPING_ZONE_LOCATION_REMOVE,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_ADD,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_CHANGE_TYPE,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_EDIT,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_REMOVE,
+	WOOCOMMERCE_SHIPPING_ZONE_REMOVE,
+} from '../../../action-types';
+
+const LOCATION_TYPES = [ 'postcode', 'state', 'country', 'continent' ];
+
+// TODO: reorder zones (do we need it for MVP?)
+
+const reducers = {};
+
+reducers[ WOOCOMMERCE_SHIPPING_ZONE_ADD ] = ( state ) => {
+	return { ...state,
+		currentlyEditingZone: {
+			id: null,
+			locations: [],
+			methods: [],
+			order: ( ( state.zones[ state.zones.length - 2 ] || {} ).order || 0 ) + 1,
```

Comment:
> I assume the length - 2 here is to account for the "Rest of the World" zone? Perhaps a comment to that effect would be helpful.

Agreed.

> Does this also mean that we're including the "Rest of the World" zone in an array that will allow ordering? Core WC handles this zone specially, do we need to as well?

I think treating all the zones as part of an array would simplify the React part of this.